pkg/oci/remote/remote.go

```diff
@@ -39,9 +40,17 @@ var (
 
 	// ErrEntityNotFound is the error that SignedEntity returns when the
 	// provided ref does not exist.
-	ErrEntityNotFound = errors.New("entity not found in registry")
+	ErrEntityNotFound = "cosign remoteGet: entity not found in registry"
 )
 
+func NewEntityNotFoundError(err error) error {
```

Comment:
At this point, it probably makes sense to make a first-class error type. Then we don't need `IsEntityNotFoundError`, we can use reflection.